### PR TITLE
feat(replay): use step numbers in favor of checkboxes

### PR DESCRIPTION
**Pull request by eliashussary**
## Summary
Minor layout changes and removal of the onboarding checkbox, to align to: https://www.figma.com/file/6e0R6eyjmohixOipp7lkq6/Specs%3A-Onboarding-v2?node-id=823-1876&t=N3Rp78p02vCHCV3t-0

Before:
image

After:
image

Relates to: https://github.com/getsentry/sentry/issues/45474

### CODE REVIEW

File: static/app/components/replaysOnboarding/sidebar.tsx [hunk 1]
```diff
@@ -8,12 +8,14 @@ import {DropdownMenu, MenuItemProps} from 'sentry/components/dropdownMenu';
 import IdBadge from 'sentry/components/idBadge';
 import LoadingIndicator from 'sentry/components/loadingIndicator';
 import useOnboardingDocs from 'sentry/components/onboardingWizard/useOnboardingDocs';
+// TODO: should be time to move this
+import {Flex} from 'sentry/components/profiling/flex';
```
Comment (billyvg) [inline static/app/components/replaysOnboarding/sidebar.tsx:12]: 😬 We spent a bunch of time in the past removing a component like this https://github.com/getsentry/sentry/pulls?q=is%3Apr+reflexbox+is%3Aclosed+

File: static/app/components/replaysOnboarding/sidebar.tsx [hunk 2]
```diff
@@ -198,8 +195,40 @@ function OnboardingContent({currentProject}: {currentProject: Project}) {
...
+  display: flex;
+  margin-right: ${space(1.5)};
+  background-color: ${p => p.theme.yellow300};
+  border-radius: 50%;
+  font-weight: bold;
+  height: ${space(4)};
+  width: ${space(4)};
+  justify-content: center;
+  align-items: center;
+`;
```
Comment (billyvg) [inline static/app/components/replaysOnboarding/sidebar.tsx:227]: I've seen this component used in other places in Sentry, is there no existing component?

File: static/app/components/replaysOnboarding/sidebar.tsx (hunk 1 above)
Comment (eliashussary) [reply static/app/components/replaysOnboarding/sidebar.tsx:12]: So I've learned... I brought this up in FE-TSC before https://www.notion.so/sentry/FE-TSC-Proposal-New-Layout-Primitive-Components-06ef641865de4f0eabcb60f5f340573f?pvs=4

i can't bring myself to keep writing styled components everytime i need some minor layout changes.

if we have really strong opinions about it, i can stop using it. realistically i don't really want a raw `Flex` component, i think we need something closer to a `Stack` and `Inline` component that has sane defaults

File: static/app/components/replaysOnboarding/sidebar.tsx (hunk 2 above)
Comment (eliashussary) [reply static/app/components/replaysOnboarding/sidebar.tsx:227]: its similar to our avatars, can't seem to find anything else

File: static/app/components/replaysOnboarding/sidebar.tsx (hunk 1 above)
Comment (billyvg) [reply static/app/components/replaysOnboarding/sidebar.tsx:12]: I agree we need higher level primitives like a `Stack` or `Inline` - I think in the mean time we just stick to styled components for now.

**Review by billyvg — APPROVED** [review]

**Pull request merged**